packages/hooks/src/useP2PAdvertList.ts

```diff
@@ -1,15 +1,20 @@
 import React from 'react';
-import { usePaginatedFetch } from '@deriv/api';
+import { useFetch } from '@deriv/api';
 import useExchangeRate from './useExchangeRate';
 
 /**
  * This custom hook returns available adverts for use with 'p2p_order_create' by calling 'p2p_advert_list' endpoint
  */
-const useP2PAdvertList = (
-    payload?: NonNullable<Parameters<typeof usePaginatedFetch<'p2p_advert_list'>>[1]>['payload']
-) => {
+const useP2PAdvertList = (payload?: NonNullable<Parameters<typeof useFetch<'p2p_advert_list'>>[1]>['payload']) => {
     const { getRate } = useExchangeRate();
-    const { data, ...rest } = usePaginatedFetch('p2p_advert_list', { payload });
+
+    const limit: number = payload?.limit || 10;
+    const [offset, setOffset] = React.useState<number>(payload?.offset || 0);
+
+    const { data, ...rest } = useFetch('p2p_advert_list', {
+        payload: { ...payload, offset, limit },
```

Comment:
would this work if we dont pass either limit or offset:

```suggestion
       payload: { ...payload, ...(offset ? { offset } : {}), ...(limit ? { limit } : {}) },
```

Reply:
This would work @ameerul-deriv, but for readability sake, which one of these 2 is more readable? Also adding conditionals to the function increases cognitive complexity, so I don't think this is necessary, as  I had to break up the logic of one of the hooks, just to reduce cognitive complexity

`payload: { ...payload, ...(offset ? { offset } : {}), ...(limit ? { limit } : {}) `

or

` payload: { ...payload, offset, limit },`

Reply:
since we're going with your implementation dont need to use the suggested one 👍